src/server/handlers/mod.rs

```diff
@@ -43,4 +56,24 @@ impl Handlers {
             Err(anyhow!("method not supported"))
         }
     }
+
+    /// Create the needed subnet managers for each subnet.
+    ///
+    /// Since we don't have a large number of subnet for now, to keep things simple,
+    /// these managers are created upon initialization.
+    ///
+    /// If the traffic received by the json rpc node increases or the number of subnets increases
+    /// significantly, we can use Connection Pooling for manage the subnet managers.
+    fn create_managers(subnets: &HashMap<String, Subnet>) -> HashMap<String, LotusSubnetManager<JsonRpcClientImpl>> {
```

Comment:
I don't understand why can't you do this when you initialize your pool/shared manager, It would be  cleaner.

Reply:
because if we init in the manager, then manager will have the `JsonRPCClient` fixed. The type should be determined in the caller.